libminifi/include/core/FlowConfiguration.h

```diff
@@ -70,7 +70,11 @@ class FlowConfiguration : public CoreComponent {
         logger_(logging::LoggerFactory<FlowConfiguration>::getLogger()) {
     controller_services_ = std::make_shared<core::controller::ControllerServiceMap>();
     service_provider_ = std::make_shared<core::controller::StandardControllerServiceProvider>(controller_services_, nullptr, configuration);
-    flow_version_ = std::make_shared<state::response::FlowVersion>("", "default", "");
+    std::string flowUrl = "", bucket_id = "default", flowId = "";
```

Comment:
String are default constructed as empty, why do we need this?

Reply:
I used auto and then let it autobox ( from the assignment ) but windows failed on the compilation of that -- I changed it without regard to the sets. I'll try to remember changing this. thanks!